### PR TITLE
Changed definition to match use of Transverse_magnification function 

### PR DESCRIPTION
**Pull request by shishir-11**
#### References to other Issues or PRs

Changes docstring and html as mentioned in issue #24438

#### Brief description of what is fixed or changed
Docstring changed to match the result given by the function . The function uses the forumla for mirror but could also be used for a lens since lack of clarification.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by HimaTeju** [comment]
@shishir-11 Only one test i.e. " test / benchmarks (pull_request) "  has failed in a couple of PRs you have made and also the same failed for me. Any idea why this could be??

**Comment by tjstienstra** [comment]
Can you change the release notes to `NO ENTRY`? Minor documentation changes do not need release notes entries, refer to [what makes a good release notes entry](https://github.com/sympy/sympy/wiki/Writing-Release-Notes#what-makes-a-good-release-notes-entry).

**Comment by shishir-11** [comment]
> Can you change the release notes to `NO ENTRY`? Minor documentation changes do not need release notes entries, refer to [what makes a good release notes entry](https://github.com/sympy/sympy/wiki/Writing-Release-Notes#what-makes-a-good-release-notes-entry).

I have changed the entry and read the guidelines for entry , thank you for the help .

File: sympy/physics/optics/utils.py [hunk 1]
```diff
@@ -671,7 +671,7 @@ def hyperfocal_distance(f, N, c):
 def transverse_magnification(si, so):
     """
 
-    Calculates the transverse magnification, which is the ratio of the
+    Calculates the transverse magnification upon reflection at mirror, which is the ratio of the
```
[the remark this answers is not in the document]
Comment (shishir-11) [reply sympy/physics/optics/utils.py]: I have corrected the grammatical error , line lengh and squashed them into a single commit . Kindly take a look at it and inform me if any more changes are needed . Thanks

**Pull request merged**